bindings/java/mongocrypt/src/main/java/com/mongodb/crypt/capi/MongoCryptImpl.java

```diff
@@ -277,34 +265,32 @@ public MongoCryptContext createExplicitEncryptionContext(final BsonDocument docu
             throwExceptionFromStatus();
         }
 
-        boolean success;
-
         if (options.getKeyId() != null) {
             try (BinaryHolder keyIdBinaryHolder = toBinary(ByteBuffer.wrap(options.getKeyId().getData()))) {
-                success = mongocrypt_ctx_setopt_key_id(context, keyIdBinaryHolder.getBinary());
-                if (!success) {
-                    MongoCryptContextImpl.throwExceptionFromStatus(context);
-                }
+                configure(() -> mongocrypt_ctx_setopt_key_id(context, keyIdBinaryHolder.getBinary()), context);
             }
         } else if (options.getKeyAltName() != null) {
             try (BinaryHolder keyAltNameBinaryHolder = toBinary(new BsonDocument("keyAltName", new BsonString(options.getKeyAltName())))) {
-                success = mongocrypt_ctx_setopt_key_alt_name(context, keyAltNameBinaryHolder.getBinary());
-                if (!success) {
-                    MongoCryptContextImpl.throwExceptionFromStatus(context);
-                }
+                configure(() -> mongocrypt_ctx_setopt_key_alt_name(context, keyAltNameBinaryHolder.getBinary()), context);
             }
         }
 
-        success = mongocrypt_ctx_setopt_algorithm(context, new cstring(options.getAlgorithm()), -1);
-        if (!success) {
-            MongoCryptContextImpl.throwExceptionFromStatus(context);
+        if (INDEXED.equalsIgnoreCase(options.getAlgorithm())) {
```

Comment:
I'm not sure I understand what's going on in this if/else block.  Why are we sometimes calling  mongocrypt_ctx_setopt_index_type and someting calling  mongocrypt_ctx_setopt_algorithm for the same option?

Reply:
Followed the logic from: https://github.com/kevinAlbs/mongo-go-driver/pull/3/files#diff-e2829099000d774cf77f3bed96c37b4afa44ce09fedddc4b58ebeb91019442beR183-R197